tachyon/crypto/hashes/sponge/poseidon2/poseidon2_params.h

```diff
@@ -0,0 +1,102 @@
+#ifndef TACHYON_CRYPTO_HASHES_SPONGE_POSEIDON2_POSEIDON2_PARAMS_H_
+#define TACHYON_CRYPTO_HASHES_SPONGE_POSEIDON2_POSEIDON2_PARAMS_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <type_traits>
+
+#include "tachyon/base/bits.h"
+#include "tachyon/base/types/always_false.h"
+#include "tachyon/math/elliptic_curves/bn/bn254/fr.h"
+#include "tachyon/math/finite_fields/finite_field_traits.h"
+
+namespace tachyon::crypto {
+
+template <typename F, size_t Width, uint64_t Alpha>
+constexpr size_t GetPoseidon2PartialRounds() {
+  using PrimeField = math::MaybeUnpack<F>;
+
+  constexpr auto SelectRounds = [](uint64_t alpha,
+                                   const std::array<size_t, 5>& rounds) {
+    switch (alpha) {
+      case 3:
+        return rounds[0];
+      case 5:
+        return rounds[1];
+      case 7:
+        return rounds[2];
+      case 9:
+        return rounds[3];
+      case 11:
+        return rounds[4];
+      default:
+        NOTREACHED();
+        return static_cast<size_t>(0);
```

Comment:
Please refer to https://google.github.io/styleguide/cppguide.html#Casting

```suggestion
        return size_t{0};
```